### PR TITLE
rest: add unit tests for middleware/usersignup

### PR DESCRIPTION
**Pull request by filariow**
Signed-off-by: Francesco Ilario <filario@redhat.com>

### CODE REVIEW

File: server/rest/middleware/usersignup_test.go [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+
+	When("sub HTTP header is not present", func() {
+		It("invokes next handler", func() {
+			// set expectations
+			c.EXPECT().List(gomock.Any(), gomock.Any()).Times(0)
+			h.EXPECT().ServeHTTP(gomock.Any(), gomock.Any()).Times(1)
+
+			// when
+			m.ServeHTTP(w, r)
+		})
```
Comment (sadlerap) [inline server/rest/middleware/usersignup_test.go:76]: This test doesn't assert anything?

Comment (filariow) [reply server/rest/middleware/usersignup_test.go]: fixed in b9dd1b2636210e1bc7f5f1437359aa60de68c9f2

**Review by sadlerap — APPROVED** [review]

**Review by dperaza4dustbit — APPROVED** [review]
lgtm

**Pull request merged**